The-SMIRNOFF-force-field-format.md

```diff
@@ -1,103 +1,288 @@
+
+Many of the specific forces are implemented as discussed in the [OpenMM Documentation](http://docs.openmm.org/latest/userguide/theory.html); see especially [Section 19 on Standard Forces](http://docs.openmm.org/latest/userguide/theory.html#standard-forces) for mathematical descriptions of these functional forms.
+In some cases, typically for consistency with the AMBER force field philosophy motivating some of the authors, we do some manipulation of parameters from these files as discussed below in "Parameter sections".
 
-## Functional forms, etc.
+**QUESTION:** Since we're defining potentials instead of forces, do we want to remove the term `Force` from these tags and instead use the simpler `<Bonds/>`, `<Angles/>`, `<Torsions/>`, etc?
 
-**Functional form**: The SMIRNOFF format specifies parameters; once specified, these are processed by the SMIRNOFF `ForceField` class and used to assign parameters to OpenMM Forces.
-This means that specific forces are generally implemented as discussed in the [OpenMM Documentation](http://docs.openmm.org/7.0.0/userguide/theory.html), see especially [Section 19 on Standard Forces](http://docs.openmm.org/7.0.0/userguide/theory.html#standard-forces) for functional forms. In some cases, typically for consistency with the AMBER force field philosophy motivating some of the authors, we do some manipulation of parameters from these files as discussed below in "Parameter sections".
+## Charge models
 
-**Charges**: In keeping with the AMBER force field philosophy, especially as implemented in small molecule force fields such as [GAFF](http://ambermd.org/antechamber/gaff.html), [GAFF2](https://mulan.swmed.edu/group/gaff.php), and [parm@Frosst](http://www.ccl.net/cca/data/parm_at_Frosst/), we at least initially treat partial charges as something to be obtained separately from the rest of the force field (bonds, angles, impropers, torsions [BAIT] and vdW terms), typically via QM calculations or a method such as Bayly's [AM1-BCC](https://dx.doi.org/10.1002/jcc.10128) approach, thus, for system setup we provide the option of specifying a charging method, though charges are not normally specified in the FFXML itself.
- For other force fields with "library"-style charges, we could introduce a new section providing specific charges via SMIRKS pattern, though this is not yet implemented. Bond charge corrections, however, are supported as we discuss below.
+SMIRNOFF supports several ways for specifying electrostatic models.
+Currently, only classical fixed point charge models are supported, but extensions to the specification will support point multipoles, point polarizable dipoles, Drude oscillators, charge equilibration methods, and so on.
+
+### Library charges
+
+A mechanism is provided for specifying library charges should be used for molecules or residues that match library charge templates.
```

Comment:
(I see your WARNING comment and note on lines 148-151, but still...) I'm not familiar with the term "library charges," but would this override any user-supplied charges already specified before calling `createSystem` with a set of OEMols and topologies? The comment below seems to suggest "yes." If so: why not have the ability to use library charges as an option?

Reply:
I'm not quite sure I understand what you're saying, @slochower . Partly I think you're saying we need to specify what library charges are, and whether they override user-supplied charges. But is there more to it? Are you saying we should be adding an argument as to whether to use these, even if provided in the force field? 

This mainly comes up when, for example, you would want to use standard charges provided by a protein when parameterizing a protein. I don't think we want an "option" for whether to use library charges or not, as if you turned it off and were, say, running a protein-ligand system with a force field which provides library charges for the protein, you would then end up with user supplied charges for the ligand and no charges for the protein. 

It will be relatively common to mix these, since current biopolymer force fields always use library charges, and small molecule force fields usually use something different (with few exceptions, e.g. GROMOS).

Reply:
I'm fairly comfortable with standard / library charges for amino acids in a protein force field and library charges for special molecules, e.g., TIP3P. But for any sort of host-guest calculations, I might want to rigorously compare the results using different charging schemes (e.g., RESP vs. `antechamber` AM1-BCC) and don't want the force field applying any modifications to my charges. I was thinking there would be a way to basically tell `createSystem` don't do anything to my charges -- I think you talk about this below, making my way there...

Reply:
One goal of SMIRNOFF is to encode a variety of schemes for experimenting with charging methods combinatorially (eg combine a QM scheme with a RESP or population analysis scheme, with or without BCCs). In the longer term, we want to allow you to do this via the SMIRNOFF spec. In the short term, our implementation should continue to support overriding some molecules with user specified charges.

Reply:
But note that the forcefield definitions we distribute MUST specify the charging scheme since that has such a huge effect on the potential, so please lobby for schemes you think are promising enough to use in production calculations!

Reply:
> One goal of SMIRNOFF is to encode a variety of schemes for experimenting with charging methods combinatorially (eg combine a QM scheme with a RESP or population analysis scheme, with or without BCCs). 

Are you saying all the information for various charging schemes will be part of SMIRNOFF and if the user specifies "I want to do a PM6 semi-empirical calculation with Millikan population analysis" (as a crazy example) and then SMIRNOFF will run appropriate OpenEye (or psi4 or whatever) commands to make it happen? I guess I'm unclear about "encod[ing] a variety of schemes" phrasing.

Reply:
@jchodera - yes, the new force fields we develop will specify a charging scheme. But other FFs can be put into this format, and people will want to experiment with charging schemes, so we also need support for user specified charges so we don't have to support (as Slochower notes) all the crazy things people might want to try.

Reply:
(And not all things might be crazy; e.g. I have a colleague who started off looking at Hirschfeld-I (updated, improved inducible hirschfeld charge model) charges and how they performed in hydration free energies; ended up ruling those out, but minimal basis set iterative stockholder charges actually look somewhat interesting -- too early to say if they'll really work out, but certainly not able to rule out at this point.)